Add architectural decision records

arhipov35 opened the pull request.

vasvjuck commented on doc/adr/001-auth-security.md:
```diff
@@ -0,0 +1,33 @@
+# 001: Implementing Authentication and Authorization Protocols to Improve Security
```
imo, it would be great if you can be more precise here in a decision section. I'd suggest to add also `rationale` section, so there is a "proper analysis" and points, why decision is right. it’d be great to see a quick comparison of other auth options you considered, with some pros and cons. that way, it’s clear the decision was well thought out.

just out of curiosity - was there a specific pattern or template you followed for structuring this ADR? If you have any resources or links to the pattern you used, would be great if you could share them!

vasvjuck commented on doc/adr/001-auth-security.md:
+1

vasvjuck commented:
quick suggestion: it’s usually helpful to prefix your PR title to indicate the type of change—like `feat`, `fix`, `docs`, or `chore`. In this case, since it's documentation-related, I'd suggest naming it something like docs: ... to make it clearer at a glance.

here is a nice article about it - https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716

arhipov35 commented:
@vasvjuck 
@KPIcoder 
I've updated ADR-001 to clarify the authentication decision and added more details on the rationale and alternatives considered.
